src/crimson/os/seastore/segment_cleaner.cc

```diff
@@ -383,20 +419,27 @@ SegmentCleaner::SegmentCleaner(
     ool_segment_seq_allocator(
       new SegmentSeqAllocator(segment_type_t::OOL)),
     gc_process(*this)
-{}
+{
+  config.validate();
+}
 
 void SegmentCleaner::register_metrics()
 {
   namespace sm = seastar::metrics;
-  stats.segment_util.buckets.resize(11);
+
+  // see also calc_utilization(seg-id)
+  // bucket 0~10 stores closed segments by usage
+  // bucket 11 stores open segments
+  // bucket 12 stores empty segments
+  stats.segment_util.buckets.resize(12);
   for (int i = 0; i < 11; i++) {
     stats.segment_util.buckets[i].upper_bound = ((double)(i + 1)) / 10;
-    if (!i) {
-      stats.segment_util.buckets[i].count = segments.get_num_segments();
-    } else {
-      stats.segment_util.buckets[i].count = 0;
-    }
+    stats.segment_util.buckets[i].count = 0;
   }
+  // NOTE: by default the segments are empty
+  stats.segment_util.buckets[11].upper_bound = 1.2;
+  stats.segment_util.buckets[11].count = segments.get_num_segments();
```

Comment:
This usage of 11 and 12 is pretty magic.  At the very least add constants, but I think segment_util should probably just have two extra members.

Reply:
Fixed in https://github.com/ceph/ceph/pull/46158/commits/eb9ffca28292e5a5f45706953d9f7e016f7591ff